framework/auth/views.py

```diff
@@ -132,12 +138,16 @@ def forgot_password_get(auth):
     :param auth: the authentication context
     :return
     """
+    #overriding the routes.py sign in url to redirect to the dashboard after login
+    context = {}
+    next_url = request.url.replace('/forgotpassword/', '/dashboard/')
+    context['login_url'] = cas.get_login_url(next_url)
```

Comment:
I'd also this whole section where you're building the context to below the `if auth.logged_in` block. it's not relevant unless the user is logged out.